src/vos/vos_dtx.c

```diff
@@ -1568,9 +1595,15 @@ vos_dtx_act_reindex(struct vos_container *cont)
 				continue;
 			}
 
-			D_ALLOC_PTR(dae);
-			if (dae == NULL)
-				D_GOTO(out, rc = -DER_NOMEM);
+			D_ASSERT(dae_df->dae_lid >= DTX_LID_RESERVED);
+			dae = lrua_allocx_inplace(cont->vc_dtx_array,
+					  dae_df->dae_lid - DTX_LID_RESERVED,
+					  dae_df->dae_epoch);
+			D_ASSERT(dae != NULL);
```

Comment:
If there is on-disk data corruption, then it is possible that two entries have have the same lid as to the second lrua_allocx_inplace() may return NULL. Similar as above, "D_ASSERT()" is not suitable.

Reply:
Ok, I can rework it to return an error.   What error?  There isn’t much we can do in such case but exiting the server is bad, I agree

Reply:
Then how about skip such entry with warning message?